app/assets/javascripts/incidents/batch_export.coffee

```diff
@@ -0,0 +1,29 @@
+handleExporting = ->
+  inSelectMode = $('table.incidents th.batch-export').is ':visible'
+  if inSelectMode
+    ids = $('.batch-export input:checked').map ->
+      $(this).data 'id'
+    .toArray()
+    alreadyExported= $('.batch-export input:checked').filter ->
+      $(this).data 'exported'
+    if alreadyExported.length > 0
+      if alreadyExported.length == 1
+        warning = '1 selected incident has already been exported.'
+      else warning = "#{alreadyExported.length} selected incidents have already been exported."
+      warning += "\nPlease doublecheck that you would like to re-export this incident."
+      unless confirm warning
+        return false
+     window.location.href = 'incidents/batch_export?' + $.param(ids: ids)
+  else
+    $('table.incidents .batch-export').show()
+    $(this).text 'Select incidents to export...'
```

Comment:
Sorry what's `this` in this scenario? I saw that somewhere else too in the button text

Reply:
The button.

Reply:
Because the function is invoked as `$('.batch-export button').click handleExporting`.